java/arcs/core/data/expression/EvaluatorParticle.kt

```diff
@@ -26,6 +27,9 @@ import arcs.sdk.ReadSingletonHandle
 import arcs.sdk.WriteCollectionHandle
 import arcs.sdk.WriteSingletonHandle
 
+fun Plan.Particle.isExpressionParticle(): Boolean =
+    this.location.isEmpty() && this.handles.values.any { it.expression != null }
+
```

Comment:
We could probably allow location to be a magic string like 'EvalutorParticle.kt'

Reply:
Interesting! If we do that, we don't need any special casing in the Allocator or the ArcHost. I will try that.